crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -76,10 +76,37 @@ impl Operations {
         self.operations.retain(|x| x.name.ne(&op_name));
     }
 
-    pub fn try_new(dir: impl AsRef<Path>, cloud_name: &str) -> Result<Self, OperationsError> {
+    pub fn try_new(
+        dir: impl AsRef<Path>,
+        cloud_name: Option<&str>,
+    ) -> Result<Self, OperationsError> {
         get_operations(dir.as_ref(), cloud_name)
     }
 
+    pub fn get_child_ops(
+        ops_dir: impl AsRef<Path>,
+    ) -> Result<HashMap<String, Self>, OperationsError> {
+        let mut child_ops: HashMap<String, Operations> = HashMap::new();
+        let child_entries = fs::read_dir(&ops_dir)
+            .map_err(|_| OperationsError::ReadDirError {
+                dir: ops_dir.as_ref().into(),
+            })?
+            .map(|entry| entry.map(|e| e.path()))
+            .collect::<Result<Vec<PathBuf>, _>>()?
+            .into_iter()
+            .filter(|path| path.is_dir())
+            .collect::<Vec<PathBuf>>();
+        for cdir in child_entries {
+            let ops = Operations::try_new(cdir.clone(), None)?;
+            if let Some(id) = cdir.iter().next_back() {
```

Comment:
Using [`cdir.file_name()`](https://doc.rust-lang.org/std/path/struct.Path.html#method.file_name) would be clearer.

Reply:
done